mongoose/mongoose.d.ts

```diff
@@ -871,7 +871,7 @@ declare module "mongoose" {
     /** Hash containing current validation errors. */
     errors: Object;
     /** The string version of this documents _id. */
-    id: string;
+    id?: string;
```

Comment:
@cyrixmorten Can you try changing MongooseDocument to an interface? It should be an interface since it doesn't actually exist on mongoose as a constructor.